Bad CPU type in executable - cache key is missing runner.arch

When running `tree-sitter/workflows/.github/workflows/package-pypi.yml@main`, it invokes this workflow with different systems including macos-15 and macos-13.
macos-15 seems to use an arm image but macos-13 doesnt.
I believe the cache key should include runner.arch, because runner.os is not reliable?

```
Run tree-sitter/setup-action/cli@v2
  with:
    tree-sitter-ref: latest
    rust-toolchain: stable
Run if [[ $TREE_SITTER_REF == latest ]]; then
  if [[ $TREE_SITTER_REF == latest ]]; then
    TREE_SITTER_REF=$(gh release view -R "$TREE_SITTER_REPO" -q .tagName --json tagName)
  fi
  printf '%s\n' >> "$GITHUB_ENV" \
      "TREE_SITTER_REF=$TREE_SITTER_REF" \
      "TREE_SITTER_REPO=$TREE_SITTER_REPO"
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    GH_TOKEN: ***
    TREE_SITTER_REF: latest
    TREE_SITTER_REPO: https://github.com/tree-sitter/tree-sitter
Run actions/cache/restore@v4
  with:
    path: /Users/runner/hostedtoolcache/tree-sitter/cli
    key: tree-sitter-cli-macOS-v0.25.8
    enableCrossOsArchive: false
    fail-on-cache-miss: false
    lookup-only: false
  env:
    TREE_SITTER_REF: v0.25.8
    TREE_SITTER_REPO: https://github.com/tree-sitter/tree-sitter
Cache hit for: tree-sitter-cli-macOS-v0.25.8
Received 5548144 of 5548144 (100.0%), 32.1 MBs/sec
Cache Size: ~5 MB (5548144 B)
/usr/local/bin/gtar -xf /Users/runner/work/_temp/1f17470f-09ac-4307-8f5a-70405c26b3de/cache.tzst -P -C /Users/runner/work/tree-sitter-mips/tree-sitter-mips --delay-directory-restore --use-compress-program unzstd
Cache restored successfully
Cache restored from key: tree-sitter-cli-macOS-v0.25.8
Run printf '/Users/runner/hostedtoolcache/tree-sitter/cli/bin\n' >> "$GITHUB_PATH"
Run config="$(tree-sitter init-config | cut -c32- | tr '\\' '/')"
/Users/runner/work/_temp/06912766-f5eb-451b-ba3f-8c8fa048cb6b.sh: line 1: /Users/runner/hostedtoolcache/tree-sitter/cli/bin/tree-sitter: Bad CPU type in executable
Error: Process completed with exit code 126.
```

--- a/cli/action.yml
+++ b/cli/action.yml
@@ -31,7 +31,7 @@ runs:
       id: cache-cli
       with:
         path: ${{runner.tool_cache}}/tree-sitter/cli
-        key: tree-sitter-cli-${{runner.os}}-${{env.TREE_SITTER_REF}}
+        key: tree-sitter-cli-${{runner.os}}-${{runner.arch}}-${{env.TREE_SITTER_REF}}
     - name: Set up Rust
       uses: actions-rust-lang/setup-rust-toolchain@v1
       if: >-
@@ -74,4 +74,4 @@ runs:
         steps.cache-cli.outputs.cache-hit != 'true'
       with:
         path: ${{runner.tool_cache}}/tree-sitter/cli
-        key: tree-sitter-cli-${{runner.os}}-${{env.TREE_SITTER_REF}}
+        key: tree-sitter-cli-${{runner.os}}-${{runner.arch}}-${{env.TREE_SITTER_REF}}

--- a/lib/action.yml
+++ b/lib/action.yml
@@ -31,7 +31,7 @@ runs:
       id: cache-lib
       with:
         path: ${{runner.tool_cache}}/tree-sitter/lib
-        key: tree-sitter-lib-${{runner.os}}-${{env.TREE_SITTER_REF}}
+        key: tree-sitter-lib-${{runner.os}}-${{runner.arch}}-${{env.TREE_SITTER_REF}}
     - name: Install tree-sitter library
       if: steps.cache-lib.outputs.cache-hit != 'true'
       working-directory: ${{runner.temp}}
@@ -77,4 +77,4 @@ runs:
         steps.cache-lib.outputs.cache-hit != 'true'
       with:
         path: ${{runner.tool_cache}}/tree-sitter/lib
-        key: tree-sitter-lib-${{runner.os}}-${{env.TREE_SITTER_REF}}
+        key: tree-sitter-lib-${{runner.os}}-${{runner.arch}}-${{env.TREE_SITTER_REF}}

